salt/config/__init__.py

```diff
@@ -2852,7 +2852,7 @@ def is_profile_configured(opts, provider, profile_name, vm_=None):
         non_size_drivers.append('linode')
 
     # If cloning on VMware, specifying image is not necessary.
-    if driver == 'vmware' and profile_key.get('image', True):
+    if driver == 'vmware' and 'image' not in list(profile_key.keys()) and 'clonefrom' in list(profile_key.keys()):
```

Comment:
Ok, I took that out - well spotted. 

Correspondingly I also enhanced a test to make sure that the error being reported is only regarding `clonefrom` (to ensure the default behaviour)